source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,206 @@
+Ian Ray, 25 July 2023.
+
+Formalizing the auxilary notion of a delta-V-complete poset which is
+sufficient for the main theorems of *link Tom de Jong's thesis*
+involving impredicativity (in the form of resizing axioms) and order theory.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import MLTT.Two-Properties
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Logic
+
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+
+module Posets.Delta-V-CompletePoset
+ (pt : propositional-truncations-exist)
+ (fe : Fun-Ext)
+  where
+
+open import Locales.Frame pt fe
+
+open import Posets.2ElementPoset pt fe
+
+module δ_𝓥-complete-poset {𝓤 𝓦 : Universe} (𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+ δ : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → (𝟙{𝓥} + P holds) → ∣ A ∣ₚ 
+ δ x y P (inl _) = x
+ δ x y P (inr _) = y
+
+ open Joins (rel-syntax A)
+
+ is-δ_𝓥-complete : 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+ is-δ_𝓥-complete = (x y : ∣ A ∣ₚ) → (x ≤[ A ] y) holds → (P : Ω 𝓥) →
+   Σ s ꞉ ∣ A ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ x y P))) holds
+
+\end{code}
+
+Future work: I would be nice to show that classically every poset is
+δ_𝓥 complete and to provide the critical examples of δ_𝓥 complete posets
+such as 𝓥-sup lattices, etc. We should also show that is-δ_𝓥-complete is
+a subsingleton.
+
+We now define the type of δ_𝓥 complete posets.
+
+\begin{code}
+
+δ_𝓥-complete-Poset : (𝓤 𝓦 𝓥 : Universe) → (𝓤 ⁺) ⊔ (𝓦 ⁺) ⊔ (𝓥 ⁺) ̇
+δ_𝓥-complete-Poset 𝓤 𝓦 𝓥 = Σ A ꞉ Poset 𝓤 𝓦 , δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 A
+
+Poset-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) → Poset 𝓤 𝓦
+Poset-δ D = pr₁ D
+
+is-δ_𝓥-complete-δ : (D : δ_𝓥-complete-Poset 𝓤 𝓦 𝓥) →
+  δ_𝓥-complete-poset.is-δ_𝓥-complete 𝓥 (Poset-δ D)
+is-δ_𝓥-complete-δ D = pr₂ D
+
+\end{code}
+
+\begin{code}
+
+module non-trivial-posets {𝓤  𝓦  𝓥 : Universe} (A : Poset 𝓤 𝓦) where
```

Comment:
It's a bit awkward that `𝓥` is a module parameter here, because it does not feature in the definition of nontriviality at all. It only becomes relevant in `wlem-lemma`. Maybe you could make it an explicit argument there?